### PR TITLE
Trial new fuzzing engine pipeline for libFuzzer.

### PR DESCRIPTION
**Pull request by oliverchang**
20% chance of running during fuzzing.
0% chance of running on other tasks.

### CODE REVIEW

File: src/python/bot/fuzzers/engine.py [hunk 1]
```diff
@@ -15,8 +15,13 @@
 
 from builtins import object
 from collections import namedtuple
+import random
+
+from system import environment
 
 _ENGINES = {}
+_TRIAL_PROBABILITY_FUZZ = 0.2
+_TRIAL_PROBABILITY_OTHERS = 0.05
```
Comment (inferno-chromium) [inline src/python/bot/fuzzers/engine.py]: Might be better to start with 0 first, fix the fuzzing failures first. breaking existing open testcases is very painful, maybe just use USE_NEW_ENGINE_IMPL on clusterfuzz.corp where changes are not visible atm and we will be refiling bugs later ?

File: src/python/bot/fuzzers/engine.py [hunk 2]
```diff
@@ -172,3 +177,16 @@ def get(name):
     return engine_class()
 
   return None
+
+
+def do_trial():
+  """Returns or not we should trial using this new implementation."""
+  if environment.get_value('USE_NEW_ENGINE_IMPL'):
```
Comment (inferno-chromium) [inline src/python/bot/fuzzers/engine.py]: Shouldn't USE_NEW_ENGINE_IMPL be used together with trial ? Is there a place you are enabling 100% ? Can also try on clusterfuzz-aarya-test project?

File: src/python/bot/fuzzers/engine.py (hunk 1 above)
Comment (oliverchang) [reply src/python/bot/fuzzers/engine.py]: Sure, I already tested this with some test jobs, but will set to 0 for now to be extra careful.

File: src/python/bot/fuzzers/engine.py (hunk 2 above)
Comment (oliverchang) [reply src/python/bot/fuzzers/engine.py]: Yes, I will be enabling this for a few jobs manually to better test things.

Comment (oliverchang) [reply src/python/bot/fuzzers/engine.py]: (To better explain my comment).

For all jobs, we will have a 20% chance of fuzzing. For certain jobs, I want to enable this completely, including reproduction etc

Comment (inferno-chromium) [reply src/python/bot/fuzzers/engine.py]: SGTM.

**Pull request merged**